microsoft-edge/devtools-guide-chromium/memory-problems/heap-snapshot-schema.md

```diff
@@ -0,0 +1,314 @@
+
+1. Open the **Memory** tool.
+
+1. Select **Heap snapshot** and then click **Take snapshot**.
+
+To find out more see [Record heap snapshots using the Memory tool](heap-snapshots.md).
+
+
+<!-- ====================================================================== -->
+## Export and view a `heapsnapshot` file
+
+Once you have recorded a heap snapshot, you can export it.
+
+1. In the **Memory** tool left sidebar, click **Save** next to the heap snapshot item you just recorded.
+
+1. Change the file extension from `.heapsnapshot` to `.json` to make it easier to open the file in a text editor.
+
+1. Open the saved file in a text editor such as Visual Studio Code.
+
+1. To facilitate reading the `heapsnapshot` file, re-format the JSON: in Visual Studio Code, right-click on the code and choose **Format document**.
+
+Note that the resulting file will be different every time you record and export a heap snapshot. Heap snapshots are dynamically generated based on the content of the web application currently being inspecting in DevTools.
+
+
+<!-- ====================================================================== -->
+## Overview of the `heapsnapshot` file format
+
+The memory used by a web application is organized in a graph by V8, the JavaScript engine used by Microsoft Edge. A graph is a data type that is composed of nodes (points on the graph) and edges (links between the points).
+
+The data in the `heapsnapshot` file represents that graph efficiently and in a way that facilitates the transfer of data as chunks between the browser process and DevTools. The `heapsnapshot` file contains a flattened representation of the relations between nodes and edges as a JSON object containing arrays of numbers and strings. Even if the file originally has a `.heapsnapshot` extension, it contains JSON.
+
+The data has two main parts:
+
+* The metadata, which contains all of the information you need in order to parse the arrays of data found in the payload to re-create the graph.
+* The payload, which contains the actual data needed to re-create the graph.
+
+Note that the format of the `heapsnapshot` file, as documented below, may change in the future as V8 and DevTools evolve. We always try to keep our documentation up to date but if you encounter a discrepancy please send us feedback on our [GitHub repository](https://github.com/MicrosoftDocs/edge-developer/issues).
+
+
+<!-- ====================================================================== -->
+## Schema of the `heapsnashot` data
+
+### Top-level structure
+
+The `heapsnapshot` JSON data contains a root object with the following properties:
+
+```json
+{
+    "snapshot": {},
+    "nodes": [],
+    "edges": [],
+    "trace_function_infos": [],
+    "trace_tree": [],
+    "samples": [],
+    "locations": [],
+    "strings": []
+}
+```
+
+| Property | Description | Format |
+| --- | --- | --- |
+| `snapshot` | Contains all the information about the format of the payload and its size | `Object` |
```

Comment:
```suggestion
| `snapshot` | Contains all the information about the format of the payload and its size. | `Object` |
```

Reply:
Is the term _payload_ adequately defined & clear?  It's not the graph but it is to be used to recreate the graph.  It's equivalent to the graph.  In the context of exchanged JSON, it's a payload.  After converting the JSON-formatted payload to a graph, a graph results.